src/mps/dmrg.jl

```diff
@@ -276,6 +276,10 @@ function dmrg(PH, psi0::MPS, sweeps::Sweeps; kwargs...)
   end
   @assert isortho(psi) && orthocenter(psi) == 1
 
+  if !isnothing(write_when_maxdim_exceeds) &&
+    (maxlinkdim(psi) > write_when_maxdim_exceeds) || (maxdim(sweeps, 1) > write_when_maxdim_exceeds)
```

Comment:
The tests are failing which makes me think this isn't quite doing the logic you're expecting. I think it makes the code clearer if you split it into nested if-statements:
```julia
if !isnothing(write_when_maxdim_exceeds)
  if (maxlinkdim(psi) > write_when_maxdim_exceeds) || (maxdim(sweeps, 1) > write_when_maxdim_exceeds)
    PH = disk(PH)
  end
end
```

Reply:
Yes I think that is a better format.